commands/install-plugin.go

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package commands
+
+import (
+	"github.com/apache/brooklyn-client/command_metadata"
+	"github.com/apache/brooklyn-client/error_handler"
+	brIO "github.com/apache/brooklyn-client/io"
+	"github.com/apache/brooklyn-client/plugin"
+	"github.com/apache/brooklyn-client/scope"
+	"github.com/codegangsta/cli"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+type InstallPlugin struct {
+	config     *brIO.Config
+	rpcService *plugin.RpcServer
+}
+
+func NewInstallPlugin(config *brIO.Config) (cmd *InstallPlugin) {
+	cmd = new(InstallPlugin)
+	cmd.config = config
+	rpcService, err := plugin.NewRpcServer(nil)
+	if err != nil {
+		error_handler.ErrorExit(err)
+	}
+	cmd.rpcService = rpcService
+	return
+}
+
+func (cmd *InstallPlugin) Metadata() command_metadata.CommandMetadata {
+	return command_metadata.CommandMetadata{
+		Name:        "install-plugin",
+		Aliases:     []string{"install", "ip"},
+		Description: "Install a plugin",
+		Usage:       "BROOKLYN_NAME install-plugin [NAME] [LOCATION]",
+		Flags:       []cli.Flag{},
+	}
+}
+
+func (cmd *InstallPlugin) Run(scope scope.Scope, c *cli.Context) {
```

Comment:
It would be great if this function could also add the newly installed plugin into the codegansta App so that e.g. it would be listed in the output along with its HelpText if you do "br --help".